src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -14475,12 +14483,39 @@ static void Main()
                 verify: Verification.Skipped,
                 expectedOutput: IncludeExpectedOutput($"{n}"));
 
+            static ImmutableArray<NamedTypeSymbol> getInlineArrayTypes(ModuleSymbol module)
+            {
+                return module.GlobalNamespace.GetTypeMembers().WhereAsArray(t => t.Name.StartsWith("<>y__InlineArray", StringComparison.Ordinal));
+            }
+
             static ImmutableArray<string> getInlineArrayTypeNames(ModuleSymbol module)
             {
-                return module.GlobalNamespace.GetTypeMembers().WhereAsArray(t => t.Name.StartsWith("<>y__InlineArray")).SelectAsArray(t => t.Name);
+                return getInlineArrayTypes(module).SelectAsArray(t => t.Name);
+            }
+
+            static bool isCompilerGeneratedAttribute(CSharpAttributeData attributeData)
+            {
+                return attributeData.AttributeClass is
+                {
+                    Name: nameof(System.Runtime.CompilerServices.CompilerGeneratedAttribute),
+                    ContainingNamespace:
+                    {
+                        Name: nameof(System.Runtime.CompilerServices),
+                        ContainingNamespace:
+                        {
+                            Name: nameof(System.Runtime),
+                            ContainingNamespace:
+                            {
+                                Name: nameof(System),
+                                ContainingNamespace.IsGlobalNamespace: true,
+                            },
+                        },
+                    },
+                };
```

Comment:
Are there helpers for this kind of thing?

Reply:
It looks like there are a few alternatives:
- `TypeSymbolExtensions.IsCompilerServicesTopLevelType()`
- Check type is equal to `module.ContainingAssembly.CorLibrary.GetTypeByMetadataName("System.Runtime.CompilerServices.CompilerGeneratedAttribute")`
- Check `type.ToTestDisplayString()` as in `SynthesizedCollections_EnsureCompilerGenerated()`